wandb/sdk/artifacts/artifact_saver.py

```diff
@@ -239,17 +243,15 @@ def before_commit() -> None:
             result_future=commit_result,
         )
 
-        # Block until all artifact files are uploaded and the
-        # artifact is committed.
-        try:
-            commit_result.result()
-        finally:
+        def done_callback(_: concurrent.futures.Future) -> None:
             step_prepare.shutdown()
+            if finalize and use_after_commit:
+                self._api.use_artifact(artifact_id)
```

Comment:
This should only be called when the future succeeded, i.e.:
```
if commit_result.exception() is None:
```

Reply:
fixed